openedx/adg/admin_dashboard/performance/helpers.py

```diff
@@ -49,10 +53,26 @@ def get_course_grades_for_users(user_ids, course_id):
         user_id__in=user_ids, course_id=course_id
     ).only('user_id', 'percent_grade', 'passed_timestamp')
 
-    return {
+    grades_dict = {
         grade.user_id: {
             'grade_percentage': f'{convert_float_point_to_percentage(grade.percent_grade)}%',
             'passed_timestamp': grade.passed_timestamp
         }
         for grade in grades
     }
+
+    if len(user_ids) != len(grades):
+        course_grade_factory = CourseGradeFactory()
+        for user_id in user_ids:
+            if grades_dict.get(user_id, None):
+                continue
+
+            course_grade = course_grade_factory.read(
+                user=User.objects.get(id=user_id), course_key=course_id
+            )
+            grades_dict[user_id] = {
+                'grade_percentage': f'{convert_float_point_to_percentage(course_grade.percent)}%',
+                'passed_timestamp': now(),
+            }
```

Comment:
Can we use a bulk_read or something similar instead of a separate get query from the db in a loop?
Even if we can't find a bulk_read for `course_grade_factory`, we should still fetch all the users in a single filter query.

Reply:
Actually this is just for the old users and won't be bottleneck in future and this can be easily removed, But yes you are right we should update.